tests/Query/BuilderTest.php

```diff
@@ -54,16 +79,105 @@ public static function provideQueryBuilderToMql(): iterable
             fn (Builder $builder) => $builder->where('foo', '>', $date),
         ];
 
-        yield 'find in array' => [
+        /** @see DatabaseQueryBuilderTest::testBasicWhereIns */
+        yield 'whereIn' => [
             ['find' => [['foo' => ['$in' => ['bar', 'baz']]], []]],
             fn (Builder $builder) => $builder->whereIn('foo', ['bar', 'baz']),
         ];
 
+        yield 'whereIn associative array' => [
+            ['find' => [['id' => ['$in' => [45582, 2, 3]]], []]],
+            fn (Builder $builder) => $builder->whereIn('id', ['issue' => 45582, 'id' => 2, 3]),
```

Comment:
Let's push this to a separate issue for now. I'm very much in favor of raising exceptions for poorly formatted values. As annoying as it may be for some users, I think it will ultimately avoid unexpected edge cases and bugs down the line.